legion/engine/physics/systems/physicssystem.cpp

```diff
@@ -21,10 +21,18 @@ namespace legion::physics
 
     void PhysicsSystem::setup()
     {
-        createProcess<&PhysicsSystem::fixedUpdate>("Physics", m_timeStep);
-
         m_broadPhase = std::make_unique<BroadphaseUniformGridNoCaching>(math::vec3(3, 3, 3));
+    }
+
+    void PhysicsSystem::update(legion::time::span deltaTime)
+    {
+        accumulator += deltaTime;
 
+        while (accumulator > m_timeStep)
```

Comment:
This could accumulate and cause long freezes if fixedUpdate() takes more than m_timeStep, might be worth to consider some kind of limit where it resets accumulation

Reply:
I'm going to clamp the number of fixed updates in one frame instead as resetting the accumulator would cause us to "lose" physics updates and I would prefer physics to be deterministic. Clamping will allow the physics engine to "catch up" after a lag spike

Reply:
Sounds good